src/ra_server.erl

```diff
@@ -1368,6 +1380,12 @@ handle_await_condition(#request_vote_rpc{} = Msg, State) ->
     {follower, State, [{next_event, Msg}]};
 handle_await_condition(#pre_vote_rpc{} = PreVote, State) ->
     process_pre_vote(await_condition, PreVote, State);
+handle_await_condition(election_timeout,
```

Comment:
This is a legit improvement. Hasn't come up since (upon a very brief review) the two conditions follower may fall into are catch-up and wal_down. Both symptomatic of a lagging follower, which prohibits it from winning anyway. 

The other two changes I am hesitant to approve.

Reply:
The two other changes were backed out.